### ISSUE TITLE
support for nostr protocol

### ISSUE THREAD

**Issue by jonassmedegaard** [joepio/atomic-data-rust#281]
Would it be sensible for Atomic Data to support the [nostr](https://github.com/fiatjaf/nostr) protocol for decentral data exchange?

I imagine that the atomic data server could act as nostr relay, accepting only data formats that it could parse semantically, and perhaps also act as a nostr client as well...

**Comment by joepio** [comment]
Looks interesting! I see a lot of similarity: focus on signing things, skipping the blockchaing thing but instead rely on more regular servers. Events are similar to [Commits](https://atomicdata.dev/classes/Commit), as standardized signed entities. Major difference is that Atomic Commits are far more restrictive and structured, instead of the plain UTF-8 content used by nostr. Both have their merits, of course. Nostr is simpler, but not as suitable for structured data.

Anyway, let's consider interoperability.

The [nostr protocol](https://github.com/fiatjaf/nostr/blob/master/nips/01.md) is pretty simple to implement. Some implementations for rust already exist. Adding this functionality to atomic-server should not be that difficult. I think most work will be reading the `filter` object and turning that into queries for Atomic (probably a couple of TPF queries would suffice), getting the hash, getting all event types working.

But I don't really see how both could be combined in a sensible way.

Some random thoughts:

I believe posts needs to be at least a bit standardized (more than UTF-8 strings) to provide a good UX + UI. It should support images, lists, links... Markdown / commonmark, or perhaps even HTML is a good place to start. Nostr does not have [types like that](https://github.com/fiatjaf/nostr/issues/33).

**Comment by jonassmedegaard** [comment]
> Atomic Commits are far more restrictive and structured, instead of the plain UTF-8 content used by nostr.

I guess you are not really talking about nostr generally above, but more specifically [event kind 1](https://github.com/fiatjaf/nostr/blob/master/nips/01.md#basic-event-kinds).

What I imagine is that Atomic would accept and serve only (made up, still to be defined as a NIP) "event kind 63" where content is JSON-LD (and likely a few other kinds too, e.g. probably also the currently defined kinds 0 and 2).

**Comment by jonassmedegaard** [comment]
Obviously Atomic would also support that other future "event kind 482" where content is JSON-AD, I just imagine that a NIP defining a more commonly accepted format might more likely be accepted in the nostr community at first ;-)

**Comment by joepio** [comment]
> I guess you are not really talking about nostr generally above, but more specifically event kind 1.

Yeah, you're right. Adding a new event kind would fix this issue. So... Let's say Atomic-Server _only_ accepts event kind 482. We would have a new way of subscribing to and getting Commits. I'll have to spend some more time thinking about this, or we could have a call sometime soon to discuss this further.

> I just imagine that a NIP defining a more commonly accepted format might more likely be accepted in the nostr community at first ;-)

Since JSON-AD is actually _also_ valid JSON-LD, too, it should still count as JSON-LD! It's just a bit more constrained.

**Comment by jonassmedegaard** [comment]
I would love a(nother) video meet - just tell me when it suits you, my calendar is wide open.

Sure, I know JSON-AD is a subset og JSON-LD - meaning that event kind 482 would be easier for you to handle, but event kind 63 would be more attractive for the nostr community to adopt
